src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -1,211 +1,195 @@
-import React from 'react';
-import noop from 'lodash/noop';
-
+import React, {
+    ReactNode,
+    MouseEventHandler,
+    useState,
+    useRef,
+    useMemo,
+    useCallback,
+    useEffect,
+} from 'react';
 import {block} from '../utils/cn';
 import {PopupPlacement} from '../Popup';
 import {Button, ButtonProps} from '../Button';
 import {Icon} from '../Icon';
 import {DotsIcon} from '../icons/DotsIcon';
 import {
+    DropdownMenuSize,
+    DropdownMenuItem,
     DropdownMenuItemMixed,
     DropdownMenuItemAction,
-    DropdownMenuItem,
-    DropdownMenuSize,
 } from './types';
 import {DropdownMenuPopup} from './DropdownMenuPopup';
 import {MenuProps} from '../Menu';
-
 import './DropdownMenu.scss';
 
 const b = block('dropdown-menu');
 
-interface DropdownMenuGeneralProps<T> {
+export type DropdownMenuProps<T> = {
+    /**
+     * Array of items.
+     * Nested arrays of items represent visually separated groups.
+     */
+    items?: (DropdownMenuItem<T> | DropdownMenuItem<T>[])[];
     /**
-     * Some context for this dropdown menu, it will be passed to any action called from this menu.
-     * Useful when rendering lists with context menus.
+     * Switcher icon.
+     */
+    icon?: ReactNode;
+    onMenuToggle?: () => void;
+    hideOnScroll?: boolean;
+    /**
+     * Applied for the switcher and the menu.
+     */
+    size?: DropdownMenuSize;
```

Comment:
May be we can move it after rest of the props, to reduce diff?

Reply:
not sure that the diff really matters here
seems better to have the most frequently used props closer to the top of the list